### PR TITLE
36: Any user can update/delete reviews

### PR DESCRIPTION
**Pull request by docia**
not author can't delete/update a review

### CODE REVIEW

**Review by m6xcom — APPROVED** [review]

**Review by Olenka-Hryk — APPROVED** [review]

**Review by Nazar-Lysak — APPROVED** [review]

File: src/controllers/user.js
Comment (vlad-khrychov) [inline src/controllers/user.js:1]: Do we need additional changed file in this PR?

File: src/services/review.js [hunk 1]
```diff
@@ -60,18 +60,29 @@ const reviewService = {
...
 
-  updateReview: async (id, currentUserId, updateData) => {
+  updateReview: async (id, currentUserId, updateData, res) => {
     const filteredUpdateData = filterAllowedFields(updateData, allowedReviewFieldsForUpdate)
 
     const review = await Review.findById(id).exec()
+    const reviewAuthor = review.author.toString()
+
+    if (currentUserId !== reviewAuthor) {
+      return res.sendStatus(403)
```
Comment (vlad-khrychov) [inline src/services/review.js:70]: `src/utils/errorsHelper.js` is useful for this.

File: src/services/review.js [hunk 2]
```diff
@@ -60,18 +60,29 @@ const reviewService = {
...
     await review.save()
   },
 
-  deleteReview: async (id) => {
+  deleteReview: async (id, currentUserId, res) => {
+    const review = await Review.findById(id).exec()
+    const reviewAuthor = review.author.toString()
+
+    if (currentUserId !== reviewAuthor) {
+      return res.sendStatus(403)
```
Comment (vlad-khrychov) [inline src/services/review.js:84]: use `src/utils/errorsHelper.js` for errors

File: src/services/review.js [hunk 3]
```diff
@@ -60,18 +60,29 @@ const reviewService = {
     })
   },
 
-  updateReview: async (id, currentUserId, updateData) => {
+  updateReview: async (id, currentUserId, updateData, res) => {
```
Comment (vlad-khrychov) [inline src/services/review.js:63]: All other services aren't receive `res`, I think this rule is appropriate for this service too.

File: src/controllers/user.js
Comment (docia) [reply src/controllers/user.js:1]: no, we don't ( :
removed an extra paragraph, so there is no change in user.js

File: src/services/review.js (hunk 3 above)
Comment (docia) [reply src/services/review.js:63]: I've used createForbiddenError function from errorsHelper and now I'm actually not using 'res' anywhere. So, please explain why do we need to pass it as a parametr?

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**